services/wfm/src/main/java/org/openkilda/wfm/topology/event/service/SwitchService.java

```
@@ -0,0 +1,138 @@
+        Switch daoSwitch = switchRepository.findBySwitchId(sw.getSwitchId());
+
+        if (daoSwitch == null) {
+            daoSwitch = new Switch();
+            daoSwitch.setSwitchId(sw.getSwitchId());
+        }
+
+        daoSwitch.setAddress(sw.getAddress());
+        daoSwitch.setHostname(sw.getHostname());
+        daoSwitch.setDescription(sw.getDescription());
+        daoSwitch.setController(sw.getController());
+        daoSwitch.setStatus(SwitchStatus.ACTIVE);
+        switchRepository.createOrUpdate(daoSwitch);
+    }
+
+    /**
+     * Set active status in the switch in neo4j.
+     *
+     * @param sw switch.
+     * @return transaction was successful or not.
+     */
+    public boolean activateSwitch(Switch sw) {
+        logger.info("Activate switch {}", sw.getSwitchId());
+        transactionManager.begin();
+        try {
+            processActivateSwitch(sw);
+            transactionManager.commit();
+        } catch (Exception e) {
+            transactionManager.rollback();
+            return false;
+        }
+        return true;
+    }
+
+    private void processActivateSwitch(Switch sw) {
+        Switch daoSwitch = switchRepository.findBySwitchId(sw.getSwitchId());
+
+        if (daoSwitch == null) {
+            daoSwitch = new Switch();
+            daoSwitch.setSwitchId(sw.getSwitchId());
+        }
+
+        daoSwitch.setStatus(SwitchStatus.ACTIVE);
+        switchRepository.createOrUpdate(daoSwitch);
+    }
+
+
+    /**
+     * Set inactive status in the switch in neo4j.
+     *
+     * @param sw switch.
+     * @return transaction was successful or not.
+     */
+    public boolean deactivateSwitch(Switch sw) {
+        logger.info("Deactivate switch {}", sw.getSwitchId());
+        transactionManager.begin();
+        try {
+            processDeactivateSwitch(sw);
+            transactionManager.commit();
+        } catch (Exception e) {
+            transactionManager.rollback();
+            return false;
```

An exception must not be swallowed without a reason.

---

Fixed